### DISCUSSION TITLE
Which of the sensors I need to add to HA's energy tab?

### DISCUSSION THREAD

**Discussion by tomerh2001** [GuyKh/iec-custom-component#149] [Q&A]
Hi, maybe this is a stupid question, but I want to add my total energy usage to HA's energy tab.
<img width="1126" alt="image" src="https://github.com/user-attachments/assets/624a403e-cbfb-4fc5-93a7-10557d307910">

I have selected this entity since it has the most fitting name for what I'm looking for, but I'm not sure if this is what I was supposed to select:
<img width="1126" alt="image" src="https://github.com/user-attachments/assets/a7e35b47-580e-4cb3-9883-aaf284368c4a">

These are the sensors I see in the meter device:
<img width="1126" alt="image" src="https://github.com/user-attachments/assets/294a51af-3b04-41de-a82b-cb6fcb788927">

**Comment by GuyKh** [comment]
First of all - there's no such thing as a bad question so thanks for asking.

I'll give you a short overview of Energy dashboard and what it can include.
In HomeAssistant, there are entities (sensors etc.) these are object that can update rapidly and very frequently.
If you'd open the details of such sensor, for example, you'll see its values over time with a very high resolution.
But - on the other hand, this means that this data should be stored and persisted into a database. So this has a potential to fill the database and increase it size.
To solve this problem - HomeAssistant (by default) cleans up the data periodically (see [Purge](https://www.home-assistant.io/integrations/recorder/#purge_keep_days) in the recorder).

On the other side of things, there's something called Long Term Statistics, the added value of this is - as the name suggest, long time data storing - not deleting the data on the purge action, but on the other hand, it's limited for up to a single value per hour.

**The Energy dashboard is capable of presenting only Long Term Statistics data, not standard sensor values**

There are multiple options of configuring LTS (Long Time Statistics) in HA - 
One being setting a sensor's `state_class` - of `measurement`, `total` or `total_increasing`  (see [here](https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics)).
Another option is to add LTS data manually in the component.

In this component I do both;
For example - sensor `..._last_meter_reading` has a `TOTAL_INCREASING` state and so it'll be presented.
In addition - for seeing an hourly (pretty) graph of usage - you'd have to chose the "non-sensor" entity - `iec:iec_meter_1234_energy_consumption` (**note the `iec:` prefix **) - this is data that is manually inserted into Long Term Statistics data table.
And just like you did in the screenshot, you can also select the "cost" entity (p.s. it doesn't take into account the fixed costs, only the consumption cost).

One more important thing to mention, the **LTS custom data is only enabled for smart meters** 

Hope this answers your question

**Comment by smarg120** [comment]
thanks for explanation. so I added both sources to the energy dashboard - both meter reading and manual source. meter reading appears and shows in the dashboard. manual source reports 0.....

**Reply by GuyKh** [reply]
If you added the component on the last couple of days, check up to 30 days back. If you'd find data there, the component is catching up
